Makefile

```diff
@@ -27,27 +27,28 @@ OPTIMIZE?=yes
 USE_LUA?=yes
 USE_BOX2D?=yes
 CCACHE?=ccache
-USE_CCACHE?=$(shell which $(CCACHE) 2>&1 > /dev/null && echo yes)
+USE_CCACHE?=$(shell which $(CCACHE) > /dev/null 2>&1 && echo yes)
```

Comment:
To my understanding, by issuing:

    USE_CCACHE=yes make

`$(CCACHE)` detection would get wrongly passed by, as an undesired side effect of explicitating an implicit default.

So to be perfect, I would consider separating the default value setting (`USE_CCACHE?=yes`) from `$(CCACHE)` detection, this latest then done **using an inconditional assignment operator `=`** (`USE_CCACHE=$(shell which $(CCACHE) >/dev/null 2>&1 && echo yes`) **only** under `ifeq ($(USE_CCACHE), yes)`.

Reply:
In case you didn't knew. If the intention is to silence output completely, [the ice cream operator does it simpler](https://superuser.com/a/335415/174140), I think.

Reply:
Hmmm. Smart. Insightful.

```
$ bash
$ which sh
/bin/sh
$ file /bin/sh
/bin/sh: symbolic link to dash
$ echo '#include <stdio.h>' >test.c
$ echo 'main(){fprintf(stdout,"stdout\n");fprintf(stderr,"stderr\n");}' >>test.c
$ gcc -Wno-implicit-int test.c -o test
$ bash -c './test &>doc0'
$ dash -c './test &>doc1'
stdout
stderr
$ more test.c doc0 doc1
::::::::::::::
test.c
::::::::::::::
#include <stdio.h>
main(){fprintf(stdout,"stdout\n");fprintf(stderr,"stderr\n");}
::::::::::::::
doc0
::::::::::::::
stderr
stdout
::::::::::::::
doc1
::::::::::::::
$ bash -c './test >doc2 2>&1'
$ dash -c './test >doc3 2>&1'
$ more doc2 doc3
::::::::::::::
doc2
::::::::::::::
stderr
stdout
::::::::::::::
doc3
::::::::::::::
stderr
stdout
$ _
```

Very right. Thanks!